core/types/block.go

```diff
@@ -611,6 +611,44 @@ func (h *Header) SanityCheck() error {
 	return nil
 }
 
+// ValidateBlockNumberOptions4337 validates the block range passed in the options parameter in the conditional transaction (EIP-4337)
+func (h *Header) ValidateBlockNumberOptions4337(minBlockNumber *big.Int, maxBlockNumber *big.Int) error {
```

Comment:
But then, how do you recommend we should send the appropriate error message? Because in this way, we can only say `block number does not fall in the provided range` which is a generalised version.

Reply:
Moved [here](https://github.com/ledgerwatch/erigon/pull/8229/commits/cc63de0c1cc00fe8a7978512bd622a94d14badb0). Let me know if this needs any changes. Thanks.